### PR TITLE
BUGFIX: update previewUrl and uris in state after editing uriPathSegment

### LINKED ISSUE #645: Editing URL path segment is not supported
**[<- must-have for 1.0.0](https://github.com/neos/neos-ui/issues/687)**

If we rename the URL path segment, move to an other document, do back to the renamed ones, throw an exception (routing issue), if reload the UI everything fine.

if the renamed node has children all children are not accessible from the tree

### PR DESCRIPTION
**Pull request by dimaip**
Fixes #645

### CODE REVIEW

**Comment by dimaip** [comment]
I'm not super happy with how I implemented it, but that's the best I could think of now. Improvement suggestions are welcomed :)

File: packages/neos-ui-redux-store/src/CR/Nodes/index.js [hunk 1]
```diff
@@ -230,7 +242,30 @@ export const reducer = handleActions({
     [CUT]: contextPath => $set('cr.nodes.clipboard', contextPath),
     [PASTE]: () => $set('cr.nodes.clipboard', ''),
     [HIDE]: contextPath => $set(['cr', 'nodes', 'byContextPath', contextPath, 'properties', '_hidden'], true),
-    [SHOW]: contextPath => $set(['cr', 'nodes', 'byContextPath', contextPath, 'properties', '_hidden'], false)
+    [SHOW]: contextPath => $set(['cr', 'nodes', 'byContextPath', contextPath, 'properties', '_hidden'], false),
+    [UPDATE_URI]: ({oldUri, newUri}) => state => {
+        const allNodes = $get('cr.nodes.byContextPath', state);
+        // Make sure to not include false positives by checking that the given segment ends either with "/" or "@"
+        const containsOldUriSegmentRegex = new RegExp(oldUri + '(/|@)');
```
Comment (skurfuerst) [inline packages/neos-ui-redux-store/src/CR/Nodes/index.js]: here, we should escape "oldUri" e.g. using https://lodash.com/docs#escapeRegExp, no?

Comment (skurfuerst) [reply packages/neos-ui-redux-store/src/CR/Nodes/index.js]: actually I would not use regexes here, as we only check for substring matching of two strings, right?

File: packages/neos-ui-redux-store/src/CR/Nodes/index.js [hunk 2]
```diff
@@ -153,6 +155,15 @@ const hide = createAction(HIDE, contextPath => contextPath);
...
 const show = createAction(SHOW, contextPath => contextPath);
 
+/**
+ * Update uris of all affected nodes after uriPathSegment of a node has changed
+ * Must update the node itself and all of its descendants
+ *
+ * @param {String} oldUri
+ * @param {String} newUri
+ */
+const updateUri = createAction(UPDATE_URI, (oldUri, newUri) => ({oldUri, newUri}));
```
Comment (skurfuerst) [inline packages/neos-ui-redux-store/src/CR/Nodes/index.js]: can we rename this to "oldUriFragment" and "newUriFragment" for consistency? :)

File: packages/neos-ui-redux-store/src/CR/Nodes/index.js (hunk 1 above)
Comment (dimaip) [reply packages/neos-ui-redux-store/src/CR/Nodes/index.js]: yep, will do so

**Comment by dimaip** [comment]
@skurfuerst thanks for the review, fixed!

**Comment by skurfuerst** [comment]
looks great :)

**Pull request merged**